pkg/gardenlet/controller/factory.go

```diff
@@ -134,8 +136,30 @@ func (f *GardenletControllerFactory) Run(ctx context.Context) error {
 		return fmt.Errorf("no seed selected by Gardenlet")
 	}
 
+	// Register Seed object if desired
+	if f.cfg.SeedConfig != nil {
+		// Convert gardenlet config to an external version
+		cfg, err := confighelper.ConvertGardenletConfigurationExternal(f.cfg)
+		if err != nil {
+			panic(fmt.Errorf("could not convert gardenlet configuration: %+v", err))
+		}
+
+		seed := &gardencorev1beta1.Seed{ObjectMeta: metav1.ObjectMeta{Name: f.cfg.SeedConfig.Name}}
+		if _, err := controllerutil.CreateOrUpdate(ctx, k8sGardenClient.Client(), seed, func() error {
+			seed.Labels = utils.MergeStringMaps(map[string]string{
+				v1beta1constants.GardenRole: v1beta1constants.GardenRoleSeed,
+			}, f.cfg.SeedConfig.Labels)
+
+			seed.Spec = cfg.SeedConfig.Spec
+			return nil
+		}); err != nil {
+			panic(fmt.Errorf("could not register seed %q: %+v", seed.Name, err))
+		}
+	}
+
 	// Read Garden secrets from any seed namespace as we assume they are synced accordingly by the Gardener Controller Manager.
 	// This requires adaption if we'll decide not to sync all Secrets for all Seeds in the future.
+	// For newly created seed, this seed namespace is about to be created by the GCM.
 	_, err = garden.ReadGardenSecrets(
```

Comment:
As these secrets are created by external component (gardener-controller-manager), we should wait for some short period of time until the secrets are created in the seed namespace. Currently without such wait the Seed registration is likely to fail in the first 1-2 attempts and succeed after these attempts.

Reply:
I have added such wait mechanism for the namespace.